app/src/main/java/com/infomaniak/mail/ui/newMessage/AiViewModel.kt

```diff
@@ -110,14 +125,18 @@ class AiViewModel @Inject constructor(
             ensureActive()
         }
 
-        handleAiResult(apiResponse, apiResponse.data?.promptMessage)
+        handleAiResult(apiResponse, apiResponse.data?.promptMessage, shouldTriggerContextTooLongInstead = false)
     }
 
     fun updateFeatureFlag(currentMailboxObjectId: String, mailboxUuid: String) = viewModelScope.launch(ioCoroutineContext) {
         sharedUtils.updateAiFeatureFlag(currentMailboxObjectId, mailboxUuid)
     }
 
-    fun isHistoryEmpty(): Boolean = history.isEmpty()
+    fun isHistoryEmpty(): Boolean = history.excludingContextMessage().isEmpty()
+
+    private fun List<AiMessage>.excludingContextMessage(): List<AiMessage> {
+        return filter { it.type == "assistant" || it.type == "user" }
+    }
```

Comment:
```suggestion
   fun isHistoryEmpty(): Boolean = history.none { it.type == "assistant" || it.type == "user" }
```

Reply:
I chose to loop twice on the list to make it easier to understand what isHistoryEmpty() does. With the `none` function call, you can't tell that you want to exclude messages of type context. We could write it `history.none { it.type != "context" }` but I'm still not sure it's easy to understand

Reply:
The fact is that is not that simple to understand with this additionnal function anyway, as it's not an enum so it's not easy to know what are the potential values of type
If you really want to keep it, as someone who had not developed the feature, I would better understand this way at least :

```suggestion
    fun isHistoryEmpty(): Boolean = history.excludingContextMessage().isEmpty()

    private fun List<AiMessage>.excludingContextMessage() = filterNotNull { it.type == "context" }
```

But once more, I think a commentary would have been fine instead of looping twice to add another function